src/hooks/useAbortController.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect } from 'react';
+
+const useAbortController = (callback: (signal: AbortSignal) => Promise<void>): void => {
+  useEffect(() => {
+    const abortController = new AbortController();
+    const { signal } = abortController;
+
+    // eslint-disable-next-line no-void
+    void callback(signal);
```

Comment:
promise from callback is ignored here, is that intended? Also I'm not sure what the purpose of `void` here, won't it work without it?